### PR TITLE
feat: Allow user to specify a format for stdout (#242)

### PR DESCRIPTION
**Pull request by yogeshtewari**
Updated code and tests to take in --format flag and pretty print formatted results on the console. Default format is tabular grid.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -48,6 +48,7 @@
     "google-cloud-spanner==3.1.0",
     "setuptools>=34.0.0",
     "jellyfish==0.8.2",
+    "tabulate==0.8.9",
```
Comment (nehanene15) [inline setup.py:51]: Where do you use tabulate? It seems like you only use the built-in pandas.DataFrame methods

File: data_validation/cli_tools.py [hunk 2]
```diff
@@ -291,6 +291,12 @@ def _configure_run_parser(subparsers):
         "-filters",
         help="Filters in the format source_filter:target_filter",
     )
+    run_parser.add_argument(
+        "--format",
+        "-format",
+        default="table",
+        help="Set the format for printing command output",
```
Comment (nehanene15) [inline data_validation/cli_tools.py]: In the 'help' field, we can also include the supported format options [table, text, csv, json]

File: data_validation/cli_tools.py [hunk 3]
```diff
@@ -291,6 +291,12 @@ def _configure_run_parser(subparsers):
         "-filters",
         help="Filters in the format source_filter:target_filter",
     )
+    run_parser.add_argument(
```
Comment (nehanene15) [inline data_validation/cli_tools.py:294]: We should add documentation on this CLI option in the README and mention that it defaults to 'table'

**Review by nehanene15 — CHANGES_REQUESTED** [review]

File: setup.py (hunk 1 above)
Comment (yogeshtewari) [reply setup.py:51]: table printing internally used tabulate

`print(result_df.to_markdown(tablefmt="fancy_grid"))`

File: README.md [hunk 4]
```diff
@@ -110,6 +110,7 @@ data-validation run
   --labels or -l KEY1=VALUE1,KEY2=VALUE2
                         (Optional) Comma-separated key value pair labels for the run.
   --verbose or -v       Verbose logging will print queries executed
+  --format or -fmt      Pretty Printing command output, Supported formats are (text, csv, json, table)
```
Comment (nehanene15) [inline README.md]:
```suggestion
  --format or -fmt      Format for stdout output, Supported formats are (text, csv, json, table)
```

File: data_validation/cli_tools.py (hunk 2 above)
Comment (yogeshtewari) [reply data_validation/cli_tools.py]: Done.

**Review by nehanene15 — APPROVED** [review]

File: data_validation/cli_tools.py (hunk 3 above)
Comment (yogeshtewari) [reply data_validation/cli_tools.py:294]: Done.

**Comment by yogeshtewari** [comment]
Merging to develop.

**Pull request merged**